app/lib/features/calendar_sync/calendar_sync.dart

```diff
@@ -91,17 +92,54 @@ Future<void> initCalendarSync({bool ignoreRejection = false}) async {
           .listen(
     eventsToSyncProvider,
     (prev, next) async {
-      if (!next.hasValue) {
+      final events = next.valueOrNull;
+      if (events == null) {
         _log.info('ignoring state change without value');
         return;
       }
-      // FIXME: we probably want to debounce this ...
-      await _refreshCalendar(calendarId, next.valueOrNull ?? []);
+      _scheduleRefresh(calendarId, events);
     },
     fireImmediately: true,
   );
 }
 
+Timer? _debounce;
+(String, List<EventAndRsvp>)? _next;
+bool _running = false;
+
+// schedules an update of the calender
+// makes sure there is only one running at a time
+void _scheduleRefresh(
+  String calendarId,
+  List<EventAndRsvp> events,
+) {
+  _debounce?.cancel(); // cancel the current debounce;
+  _debounce = Timer(const Duration(seconds: 3), () {
+    _debounce = null;
+    _refreshLoop();
+  });
+  _next = (calendarId, events);
+}
```

Comment:
The actual core to ensure the synchronisity quickly explained:

1. we have a timer (`_debounce`) we reset on every schedule call to issue in 3 seconds time - this allows us to wait for a bunch of things to come in at first before doing the actual update
2. then we save the current values.
3. when the timer triggers it resets the current timer and starts the refreshloop

following below.